tests/test_create.py

```diff
@@ -1063,6 +1063,42 @@ def side_effect(self, url, **kwargs):
                 # The mock should have been called with our local channel URL though.
                 assert result_dict.get('local_channel_seen')
 
+    def test_create_from_extracted(self):
+        # Test that we can link a package from its extracted directory
+        # even if the tarball is no longer available.
+        pkgs_dir = PackageCache.first_writable().pkgs_dir
+        mkdir_p(pkgs_dir)
+        pkgs_dir_hold = pkgs_dir + '_hold'
```

Comment:
Seems to work well, I updated the PR accordingly and also took your also review comments into account.

Another thing that occurred to me is that quite a few of these integration tests primarily test logic inside conda and would provide just as much value if they operated off a small local channel with just a few small packages in it instead of the actual anaconda or conda-forge channels. This could speed up many of these tests quite a bit. Obviously beyond the scope of this PR, but might be worth considering.